src/arkade-lightning.ts

```diff
@@ -540,58 +540,123 @@ export class ArkadeLightning {
+            boltzSignedRefundTx,
+            signedRefundTx
+        );
+        const combinedSignedCheckpointTx = combineTapscriptSigs(
+            boltzSignedCheckpointTx,
+            signedCheckpointTx
+        );
+
+        // get server to sign its part of the combined transaction
         const { arkTxid, finalArkTx, signedCheckpointTxs } =
             await this.arkProvider.submitTx(
-                base64.encode(signedArkTx.toPSBT()),
-                checkpoints.map((c) => base64.encode(c.toPSBT()))
+                base64.encode(combinedSignedRefundTx.toPSBT()),
+                [base64.encode(unsignedCheckpointTx.toPSBT())]
             );
 
-        // verify the server signed the transaction with correct key
-        if (
-            !this.validFinalArkTx(
-                finalArkTx,
-                serverXOnlyPublicKey,
-                vhtlcScript.leaves
-            )
-        ) {
-            throw new Error("Invalid final Ark transaction");
+        // verify the final tx is properly signed
+        const tx = Transaction.fromPSBT(base64.decode(finalArkTx));
+        const inputIndex = 0;
+        const requiredSigners = [
+            hex.encode(ourXOnlyPublicKey),
+            hex.encode(boltzXOnlyPublicKey),
+            hex.encode(serverXOnlyPublicKey),
+        ];
+
+        if (!this.verifySignatures(tx, inputIndex, requiredSigners)) {
+            throw new Error("Invalid refund transaction");
         }
 
-        const finalCheckpoints = await Promise.all(
-            signedCheckpointTxs.map(async (c) => {
-                const tx = Transaction.fromPSBT(base64.decode(c), {
-                    allowUnknown: true,
-                });
-                const signedCheckpoint = await vhtlcIdentity.sign(tx, [0]);
-                return base64.encode(signedCheckpoint.toPSBT());
-            })
+        // validate we received exactly one checkpoint transaction
+        if (signedCheckpointTxs.length !== 1) {
+            throw new Error(
+                `Expected one signed checkpoint transaction, got ${signedCheckpointTxs.length}`
+            );
+        }
+
+        // combine the checkpoint signatures
+        const serverSignedCheckpointTx = Transaction.fromPSBT(
+            base64.decode(signedCheckpointTxs[0])
+        );
+
+        const finalCheckpointTx = combineTapscriptSigs(
+            combinedSignedCheckpointTx,
+            serverSignedCheckpointTx
         );
-        await this.arkProvider.finalizeTx(arkTxid, finalCheckpoints);
+
+        // finalize the transaction
+        await this.arkProvider.finalizeTx(arkTxid, [
+            base64.encode(finalCheckpointTx.toPSBT()),
+        ]);
 
         // update the pending swap on storage if available
-        const finalStatus = await this.getSwapStatus(pendingSwap.id);
         await this.savePendingSubmarineSwap({
             ...pendingSwap,
-            status: finalStatus.status,
+            refundable: true,
+            refunded: true,
```

Comment:
sounds counterintuitive to be both refunded and refundable no ? In theory can be refunded only one time or am I missing something?